integration_tests/ffi_bindings.ts

```diff
@@ -0,0 +1,166 @@
+/**
+ * FFI bindings for cpp-bindings-linux shared library
+ */
+
+// Error callback type: (status_code: number, message: string) => void
+export type ErrorCallback = (statusCode: number, message: string) => void;
+
+// FFI symbol definitions
+export type SerialLib = {
+    readonly serialOpen: (
+        port: Deno.PointerValue,
+        baudrate: number,
+        dataBits: number,
+        parity: number,
+        stopBits: number,
+        errorCallback: Deno.PointerValue | null,
+    ) => bigint;
+
+    readonly serialClose: (
+        handle: bigint,
+        errorCallback: Deno.PointerValue | null,
+    ) => number;
+
+    readonly serialRead: (
+        handle: bigint,
+        buffer: Deno.PointerValue,
+        bufferSize: number,
+        timeoutMs: number,
+        multiplier: number,
+        errorCallback: Deno.PointerValue | null,
+    ) => number;
+
+    readonly serialWrite: (
+        handle: bigint,
+        buffer: Deno.PointerValue,
+        bufferSize: number,
+        timeoutMs: number,
+        multiplier: number,
+        errorCallback: Deno.PointerValue | null,
+    ) => number;
+};
```

Comment:
`Deno.PointerValue` should already be a union type containing `null`

```suggestion
export type SerialLib = {
    serialOpen: (
        port: Deno.PointerValue,
        baudrate: number,
        dataBits: number,
        parity: number,
        stopBits: number,
        errorCallback: Deno.PointerValue,
    ) => bigint;

    serialClose: (
        handle: bigint,
        errorCallback: Deno.PointerValue,
    ) => number;

    serialRead: (
        handle: bigint,
        buffer: Deno.PointerValue,
        bufferSize: number,
        timeoutMs: number,
        multiplier: number,
        errorCallback: Deno.PointerValue,
    ) => number;

    serialWrite: (
        handle: bigint,
        buffer: Deno.PointerValue,
        bufferSize: number,
        timeoutMs: number,
        multiplier: number,
        errorCallback: Deno.PointerValue,
    ) => number;
} as const;
```

Reply:
Resolved but did not change?

Reply:
null is still valid and could be the type no? for errorCallback

Reply:
not valid syntax: `Cannot find name 'as'.deno-ts(2304)`